test/test_git.py

```diff
@@ -364,3 +374,11 @@ def counter_stderr(line):
 
         self.assertEqual(count[1], line_count)
         self.assertEqual(count[2], line_count)
+
+    def test_execute_kwargs_set_agrees_with_method(self):
+        parameter_names = inspect.signature(cmd.Git.execute).parameters.keys()
+        self_param, command_param, *most_params, extra_kwargs_param = parameter_names
+        self.assertEqual(self_param, "self")
+        self.assertEqual(command_param, "command")
+        self.assertEqual(set(most_params), cmd.execute_kwargs)  # Most important.
+        self.assertEqual(extra_kwargs_param, "subprocess_kwargs")
```

Comment:
Or `git.cmd.execute_kwargs` could be generated in a manner similar to this, and this test, the need to manually update it, and the note in the `Git.execute` docstring about that need, could all be done away with. Maybe something like this:

```python
execute_kwargs = inspect.signature(Git.execute).parameters.keys() - {"self", "command", "subprocess_kwargs"}
```

Right now it is defined very high up in the module, even higher than `__all__`, and this suggests that it may be intended to be available for static inspection. But I don't think any tools will statically inspect a *set* of strings like that (plus, static analysis tools *can* examine the parameters of `Git.execute`... though the incomplete lists of parameters in the `@overload`-decorated stubs that precede it confuse the situation somewhat).

`git.cmd.__all__` contains only `"Git"` and I hope that means code that uses GitPython should not be using `git.cmd.execute_kwargs` or relying on its presence. If possible, perhaps it could be made more explicitly private (`_execute_kwargs`, or if it needs to remain statically defined, maybe `_EXECUTE_KWARGS`) or, even better, removed altogether if the `Git.execute` method's arguments can be inspected efficiently enough without it where `execute_kwargs` is currently used. On the other hand, people may have come to depend on it even though the presence of `__all__` that omits it means no one *should* have depended on it.

Anyway, I do not think any of the changes I suggest in this comment need to be made in this pull request. But I wanted to mention the issue just in case.

Reply:
Removing `execute_kwargs` seems like a reduction of complexity, which would always be a valuable reduction of maintenance costs should changes need to be made.

As `execute_kwargs` was never advertised in `__all__` I'd think that it's fair to say that those who depend on it nonetheless new the risk. I think the same argument is valid knowing that nothing is ever truly private, everything can be introspected if one truly wants to, yet it's something one simply has to ignore in order to be able to make any changes to python software once released.